core/app/[locale]/(default)/account/[tab]/_components/addresses-content/edit-address.tsx

```diff
@@ -0,0 +1,300 @@
+'use client';
+
+import { useRouter } from 'next/navigation';
+import { useTranslations } from 'next-intl';
+import { useRef, useState } from 'react';
+import { useFormStatus } from 'react-dom';
+import ReCaptcha from 'react-google-recaptcha';
+
+import {
+  createFieldName,
+  FieldNameToFieldId,
+  FieldWrapper,
+  Picklist,
+  PicklistOrText,
+  Text,
+} from '~/app/[locale]/(default)/login/register-customer/_components/register-customer-form/fields';
+import { Link } from '~/components/link';
+import { Button } from '~/components/ui/button';
+import { Field, Form, FormSubmit } from '~/components/ui/form';
+import { Message } from '~/components/ui/message';
+
+import { deleteAddress } from '../../_actions/delete-address';
+import { updateAddress } from '../../_actions/update-address';
+import { useAccountStatusContext } from '../account-status-provider';
+import { Modal } from '../modal';
+
+import { AddressFields, Countries } from './add-address';
+import {
+  createCountryChangeHandler,
+  createTextInputValidationHandler,
+} from './address-field-handlers';
+import { Address } from './customer-edit-address';
+
+interface FormStatus {
+  status: 'success' | 'error';
+  message: string;
+}
+
+type CountryStates = Countries[number]['statesOrProvinces'];
+type FieldUnionType = Exclude<
+  keyof typeof FieldNameToFieldId,
+  'email' | 'password' | 'confirmPassword' | 'exclusiveOffers' | 'currentPassword'
+>;
+
+const isExistedField = (name: unknown): name is FieldUnionType => {
+  if (typeof name === 'string' && name in FieldNameToFieldId) {
+    return true;
+  }
+
+  return false;
+};
+
+interface SumbitMessages {
+  messages: {
+    submit: string;
+    submitting: string;
+  };
+}
+
+const SubmitButton = ({ messages }: SumbitMessages) => {
+  const { pending } = useFormStatus();
+
+  return (
+    <Button
+      className="relative items-center px-8 py-2 md:w-fit"
+      loading={pending}
+      loadingText={messages.submitting}
+      variant="primary"
+    >
+      {messages.submit}
+    </Button>
+  );
+};
```

Comment:
@bc-alexsaiannyi can Button component to have a loading state used here instead?